src/roles/check_database_connection/tasks/main.yaml

```diff
@@ -0,0 +1,19 @@
+---
+- name: Check Foreman database connectivity and credentials
+  community.postgresql.postgresql_ping:
+    login_host: "{{ foreman_database_host }}"
+    login_user: "{{ foreman_database_user }}"
+    login_password: "{{ foreman_database_password }}"
+    login_db: "{{ foreman_database_name }}"
+  register: check_database_connection_ping_result
+  when: (foreman_database_host | default("localhost")) not in ["localhost", "127.0.0.1", "::1"]
```

Comment:
you can check `database_mode == 'external'` instead, similar to here:

https://github.com/theforeman/foremanctl/blob/a56ab1daa893df57fda45cbb8a2d5c788b9da420/src/playbooks/deploy/deploy.yaml#L61-L63